api/controller.py

```diff
@@ -515,39 +559,30 @@ def sync(self):
             self.circulation, patron)
         return feed_response(feed, cache_for=None)
 
-    def borrow(self, data_source, identifier_type, identifier, mechanism_id=None):
+    def borrow(self, identifier_type, identifier, mechanism_id=None):
         """Create a new loan or hold for a book.
 
         Return an OPDS Acquisition feed that includes a link of rel
         "http://opds-spec.org/acquisition", which can be used to fetch the
         book or the license file.
         """
+        patron = flask.request.patron
 
-        # Turn source + identifier into a LicensePool
-        pool = self.load_licensepool(data_source, identifier_type, identifier)
-        if isinstance(pool, ProblemDetail):
-            # Something went wrong.
-            return pool
-
-        # Find the delivery mechanism they asked for, if any.
-        mechanism = None
-        if mechanism_id:
-            mechanism = self.load_licensepooldelivery(pool, mechanism_id)
-            if isinstance(mechanism, ProblemDetail):
-                return mechanism
-
-        if not pool:
-            # I've never heard of this book.
+        # TODO: First make sure the patron does not already have this book
+        # on loan.
```

Comment:
This seems like an unusually important TODO. After landing this branch we should file an issue so that we don't lose track of it.